timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/AsyncLockService.java

```diff
@@ -39,6 +45,7 @@
     private final HeldLocksCollection heldLocks;
     private final AwaitedLocksCollection awaitedLocks;
     private final ImmutableTimestampTracker immutableTsTracker;
+    private final UUID serviceId;
```

Comment:
at this point I've seen enough UUIDs in this PR that I'd be tempted to have a wrapper type LeadershipId? which just wraps a UUID and has a random() method? Generally prefer wrapping types, just because they make clear what the thing's meant to be in all contexts.